### DISCUSSION TITLE
How to use the older schematics generator?

### DISCUSSION THREAD

**Discussion by ruiaraujo012** [cyrilletuzi/vscode-angular-schematics#915] [Q&A] [answered]
How can I use the older schematics generator? I'm still using an older version of Angular, and I'm missing the default behavior of the older generator, like auto module declaration.

**Comment by cyrilletuzi** [comment, marked as answer]
Hi,

All Angular LTS versions now support standalone components. And while NgModules are still there for backward compatibility, Angular 17 has made it very clear that standalone components are the new way to go (with Angular CLI generating a standalone project and standalone components by default, the new Angular.dev documentation using standalone components, Angular Material being migrated to standalone components, Ionic 7 having migrated to standalone components,....).

So NgModules can now be considered as a legacy feature.

At the same time, while this extension is very popular, with nearly 1 million installations, it is *not* a tool developed by the Angular team or affiliated to Google in any way. It is months of *unpaid* work by a single contributor.

Maintaining legacy features has a cost. The cost companies making money do not invest to do migrations should not be the charge of unpaid contributors.

So that is why the legacy schematics has been moved to the [Pro edition](https://cyrilletuzi.gumroad.com/l/schematicspro). Feel free to enjoy legacy and advanced features and support a contributor at the same time! 💖

**Reply by ruiaraujo012** [reply]
Hi, you're right, I then research a bit about it and I agree with you, we should use standalone components.

Thank you for your time :)
